### PR TITLE
docs: update readme

### LINKED ISSUE #1: I'd like to learn "why" in the README.
In the README, I'd like to see the following questions answered:
Why does this repo exist?
Why should I use it?
How should I use it?
Can I find an example of someone using it?

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,134 @@
...
+
+1. Prepare the Docker image for the test.
+2. Run Docker with appropriate flags to run the test.
+3. Upload the newly built Docker image.
+
+We'll discuss each steps in detail.
+
+### Prepare the Docker image
+
+1. Try to download a Docker image specified by `TRAMPOLINE_IMAGE`
```
Comment (chingor13) [inline README.md:29]: I assume these are mutually exclusive. Does one have higher priority?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,134 @@
...
+   environment variable.
+2. If `TRAMPOLINE_DOCKERFILE` environment variable is specified, it
+   will build the Docker image from that Dockerfile. If there's a
+   downloaded Docker image, Trampoline V2 uses the downloaded image as
+   a cache.
+
+We recomnend you have the Dockerfile in the repo and specify
```
Comment (chingor13) [inline README.md]:
```suggestion
We recommend you have the Dockerfile in the repo and specify
```

Comment (SurferJeffAtGoogle) [inline README.md]: > the repo

which repo?

File: README.md [hunk 3]
```diff
@@ -1,4 +1,134 @@
...
+Dockerfile.
+
+### Run Docker with appropriate flags to run the test
+
+* Mounting the source at /workspace
+* Add appropriate environment variables
+* Use invoker's uid and the Docker gid on the host.
+* Runs a command specified by `TRAMPOLINE_BUILD_FILE` environment
+  variable.
+* Trampoline V2 will exit with the same exit code as the build file.
```
Comment (chingor13) [inline README.md]: nit: some of these are phrases, while the last is a full sentence. Presumably the subject is "Trampoline V2" for all the bullet points

File: README.md [hunk 4]
```diff
@@ -1,4 +1,134 @@
...
+
+#### Environment variables which will be passed down into the container
```
Comment (chingor13) [inline README.md:51]: To clarify, these are only available in the container if they are available to the Kokoro v2 script?

If I run this locally outside of Kokoro, I won't get any of these environment variables?

File: README.md [hunk 5]
```diff
@@ -1,4 +1,134 @@
...
+  (true|false) Indicates we're running trampoline_v2.
+* `KOKORO_BUILD_NUMBER`
+* `KOKORO_BUILD_ID`
+* `KOKORO_JOB_NAME`
+* `KOKORO_GIT_COMMIT`
+* `KOKORO_GITHUB_COMMIT`
+* `KOKORO_GITHUB_PULL_REQUEST_NUMBER`
+* `KOKORO_GITHUB_PULL_REQUEST_COMMIT`
+* `KOKORO_GITHUB_COMMIT_URL`
+* `KOKORO_GITHUB_PULL_REQUEST_URL`
```
Comment (chingor13) [inline README.md]: suggestion: maybe a table with expected format/examples would look nice

File: README.md [hunk 6]
```diff
@@ -1,4 +1,134 @@
...
+
+#### Environment variables which will be passed down into the container
+
+The following environment variables are passed down into the container.
+
+* `RUNNING_IN_CI`
+  Tells scripts whether they are running as part of CI or not.
+* `TRAMPOLINE_CI`
+  Indicates which CI system we're in.
+* `TRAMPOLINE_V2`
```
Comment (SurferJeffAtGoogle) [inline README.md]: How about making the version the value.  In other words,
`export TRAMPOLINE_VERSION=2`

File: README.md [hunk 7]
```diff
@@ -1,4 +1,134 @@
...
+By design, all the parameters to Trampoline V2 are passed by
+environment variables.
+
+There are two required environment variables:
+* `TRAMPOLINE_IMAGE`: The docker image to use. You can use Docker
+  image URL on anywhere, but if you also want to upload the docker
+  image, use an URL on Google Container Registry
+  (gcr.io/PROJ/IMAGE_NAME).
+* `TRAMPOLINE_BUILD_FILE`: The script to run in the docker
+  container. It should be a relative path from the git root.
```
Comment (chingor13) [inline README.md:105]: Similar to above, both are not required - you need one of them set?

File: README.md [hunk 8]
```diff
@@ -1,4 +1,134 @@
...
+### Hello World!
+
+Here is a minimum example of the usage.
+
+```sh
+$ mkdir tmp
+$ echo '#!/bin/sh' > tmp/hello.sh
+$ echo 'echo "Hello World!"' >> tmp/hello.sh
+chmod +x tmp/hello.sh
+TRAMPOLINE_IMAGE=bash TRAMPOLINE_BUILD_FILE=tmp/hello.sh ./trampoline_v2.sh
```
Comment (chingor13) [inline README.md]:
```suggestion
$ chmod +x tmp/hello.sh
$ TRAMPOLINE_IMAGE=bash TRAMPOLINE_BUILD_FILE=tmp/hello.sh ./trampoline_v2.sh
```

File: README.md [hunk 9]
```diff
@@ -1,4 +1,134 @@
...
+```
+
+If you see "Hello World!", it's working.
+
+### Real world examples
+
+* `tests/python` directory has example tests and a build script. Currently it runs on travis. See `.travis.yml` file for configuration.
```
Comment (SurferJeffAtGoogle) [inline README.md:138]: We've abandoned Travis.  Kokoro or Github actions are preferred alternatives.

File: README.md [hunk 10]
```diff
@@ -1,4 +1,134 @@
...
+   a cache.
+
+We recomnend you have the Dockerfile in the repo and specify
+`TRAMPOLINE_DOCKERFILE`. This will allow you to create a single pull
+request containing changes in tests as well as changes in the
+Dockerfile.
+
+### Run Docker with appropriate flags to run the test
+
+* Mounting the source at /workspace
```
Comment (chingor13) [inline README.md]:
```suggestion
* Mounting the current working directory at /workspace
```

File: README.md (hunk 1 above)
Comment (tmatsuo) [reply README.md:29]: It download the image

then
if `TRAMPOLINE_DOCKERFILE` is specified, it always build the image, utilizing the downloaded image as a cache.

Both can happen.

File: README.md (hunk 3 above)
Comment (tmatsuo) [reply README.md]: I'll make them all sentences.

File: README.md (hunk 4 above)
Comment (tmatsuo) [reply README.md:51]: > To clarify, these are only available in the container if they are available to the Kokoro v2 script?

Yes

> If I run this locally outside of Kokoro, I won't get any of these environment variables?

You'll get the envvars inside the container everywhere.

File: README.md (hunk 5 above)
Comment (tmatsuo) [reply README.md]: will do

File: README.md (hunk 7 above)
Comment (tmatsuo) [reply README.md:105]: I guess you're talking about `TRAMPOLINE_DOCKERFILE`?

These envvars are mandatory now.

File: README.md (hunk 10 above)
Comment (tmatsuo) [reply README.md]: Actualy it's not the current working directory. Now it tries to find a git root and mount the whole git root.

File: README.md [hunk 11]
```diff
@@ -1,4 +1,134 @@
...
-This is a shell script for running another script in a Docker
-container.
+This repository contains `trampoline_v2.sh`, a shell script for
+running another script in a Docker container. Primary use case of this
+shell script is to use it as a test driver on various CI systems. By
+utilizing Docker, your tests will be very portable so that you can run
+them on any CI systems as well as on your workstations with minimum
+efforts.
+
+## trampoline_v2.sh
```
Comment (SurferJeffAtGoogle) [inline README.md:17]: After reading the whole script top to bottom, I'm still having a little difficulty understanding its purpose.

Could you add a section discussing this script's inputs and outputs, at a very general level?

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (tmatsuo) [reply README.md]: works for me

File: README.md (hunk 2 above)
Comment (tmatsuo) [reply README.md]: your repo

File: README.md (hunk 9 above)
Comment (tmatsuo) [reply README.md:138]: We can add them in the future PR and add examples here too

File: README.md (hunk 7 above)
Comment (chingor13) [reply README.md:105]: Oops, yeah misread.

File: README.md (hunk 2 above)
Comment (tmatsuo) [reply README.md]: clarified a bit

File: README.md (hunk 6 above)
Comment (tmatsuo) [reply README.md]: Done (next commit)

File: README.md (hunk 11 above)
Comment (tmatsuo) [reply README.md:17]: I updated the first section of the README. WDYT?

**Review by tmatsuo — COMMENTED** [review]
@chingor13 @SurferJeffAtGoogle 

Thanks for the review. I think addressed your comments and answered some questions.

PTAL

File: README.md (hunk 11 above)
Comment (SurferJeffAtGoogle) [reply README.md:17]: Nice.  That new paragraph forms a clear image in my brain.

File: README.md (hunk 9 above)
Comment (SurferJeffAtGoogle) [reply README.md:138]: Ok.  Please create an issue for it.

**Review by SurferJeffAtGoogle — APPROVED** [review]

File: README.md (hunk 9 above)
Comment (tmatsuo) [reply README.md:138]: Filed #6

File: README.md (hunk 10 above)
Comment (tmatsuo) [reply README.md]: I clarified what will be mounted at /workspace.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**